sematic/ui/packages/common/src/typeViz/views/hugging_face_reference.tsx

```diff
@@ -0,0 +1,106 @@
+import { OpenInNew } from "@mui/icons-material";
+import { Button, Tooltip } from "@mui/material";
+import { useTextSelection } from "@sematic/common/src/hooks/textSelectionHooks";
+import { ViewComponentProps} from "src/typeViz/common";
+
+type Reference = {
+    owner: string | null;
+    repo: string;
+    commit_sha: string | null;
+    subset?: string | null | undefined;
+}
+
+export function HuggingFaceModelReferenceShortView(props: ViewComponentProps) {
+    const { valueSummary } = props;
+    const { values } = valueSummary;
+
+    return <HuggingFaceButton
+        reference={values}
+        objectTypePrefix=""
+        short={true}
+    />;
+}
+
+export function HuggingFaceDatasetReferenceShortView(props: ViewComponentProps) {
+    const { valueSummary } = props;
+    const { values } = valueSummary;
+
+    return <HuggingFaceButton
+        reference={values}
+        objectTypePrefix="/datasets"
+        short={true}
+    />;
+}
+
+export function HuggingFaceModelReferenceValueView(props: ViewComponentProps) {
+    const { valueSummary } = props;
+    const { values } = valueSummary;
+  
+    return <HuggingFaceButton
+        reference={values}
+        objectTypePrefix=""
+        short={false}
+    />;
+
+}
+
+export function HuggingFaceDatasetReferenceValueView(props: ViewComponentProps) {
+    const { valueSummary } = props;
+    const { values } = valueSummary;
+
+    return <HuggingFaceButton
+        reference={values}
+        objectTypePrefix="/datasets"
+        short={false}
+    />;
+}
+
+function HuggingFaceButton(props: {
+    reference: Reference,
+    objectTypePrefix: string,
+    short: boolean
+}) {
+    const { reference, objectTypePrefix, short } = props;
+    const { owner, repo, subset, commit_sha } = reference;
+
+    const ownerSegment = owner ? "/" + owner : "";
```

Comment:
Please check whether the calculations could be saved with `useMemo`.

Reply:
done